avy-zap.el

```diff
@@ -184,13 +184,12 @@ Otherwise, don't rebind."
         (window-start (&optional window) (point))
       (if (or (equal avy-zap-function #'kill-region)
               (equal avy-zap-function #'delete-region))
-	  (progn
-            (call-interactively 'avy-goto-char)
-	    (when (avy-zap--xor (<= start (point)) zap-up-to-char-p)
-              (forward-char))
-	    (funcall avy-zap-function start (point)))
+	  (when (and (call-interactively 'avy-goto-char)
+		     (avy-zap--xor (< start (point)) zap-up-to-char-p))
+            (forward-char))
+	(funcall avy-zap-function start (point)))
```

Comment:
This should be in the body of `when`.

Reply:
Not really. Actually avy-goto-char returns null only when empty RET or C-g, so the cursor has not moved in those situations, which means that the region is empty (point == mark) and calling kill-region or delete-region with empty region does nothing.

Reply:
I understand that part, but putting `funcall` onside `when` means it is now in the "else" part of the if expression, and you should only call `avy-zap-function` when `avy-goto-char` returns non-nil, right?

Reply:
You were right I fixed it in a different way now.